### PR TITLE
HADOOP-16376: Override access() to no-up

### PR DESCRIPTION
**Pull request by DadanielZ**
- Gen1 driver override` FileSystem.access()` and forward it to storage service, but ABFS doesn't have this and is having some hive permission issue. As a short term fix, ABFS could override this to be a no-op.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 46 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1182 | trunk passed |
| +1 | compile | 29 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 33 | trunk passed |
| +1 | shadedclient | 785 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | trunk passed |
| 0 | spotbugs | 55 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 52 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 27 | the patch passed |
| +1 | compile | 24 | the patch passed |
| +1 | javac | 24 | the patch passed |
| +1 | checkstyle | 14 | the patch passed |
| +1 | mvnsite | 28 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 826 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 20 | the patch passed |
| +1 | findbugs | 55 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 74 | hadoop-azure in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 3362 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-971/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/971 |
| JIRA Issue | HADOOP-16376 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux eedfdfd02367 4.4.0-144-generic #170~14.04.1-Ubuntu SMP Mon Mar 18 15:02:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ae4143a |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-971/1/testReport/ |
| Max. process+thread count | 329 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-971/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by DadanielZ** [comment]
All tests passed my US-west account:
xns account:
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
Tests run: 393, Failures: 0, Errors: 0, Skipped: 25
Tests run: 190, Failures: 0, Errors: 0, Skipped: 23

non-xns account:
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
Tests run: 393, Failures: 0, Errors: 0, Skipped: 207
Tests run: 190, Failures: 0, Errors: 0, Skipped: 15

File: hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java [hunk 1]
```diff
@@ -843,6 +844,24 @@ public AclStatus getAclStatus(final Path path) throws IOException {
...
+   * @param path Path to check
+   * @param mode type of access to check
+   * @throws AccessControlException        if access is denied
+   * @throws java.io.FileNotFoundException if the path does not exist
+   * @throws IOException                   see specific implementation
+   */
+  @Override
+  public void access(final Path path, FsAction mode) throws IOException {
+    // make it no op to unblock hive permission issue for now.
+    // In future it should do same thing as AdlFileSystem.
```
Comment (jiwq) [inline hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java]: We should better add todo comments.

Comment (DadanielZ) [reply hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java]: @jiwa thanks, added

**Comment by DadanielZ** [comment]
@steveloughran @ThomasMarquardt @ajfabbri FYI this is a short term fix, will provide a long term fix in future as it depends on changes in storage service.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1070 | trunk passed |
| +1 | compile | 32 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 36 | trunk passed |
| +1 | shadedclient | 714 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 18 | trunk passed |
| 0 | spotbugs | 49 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 48 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 26 | the patch passed |
| +1 | compile | 25 | the patch passed |
| +1 | javac | 25 | the patch passed |
| -0 | checkstyle | 12 | hadoop-tools/hadoop-azure: The patch generated 1 new + 1 unchanged - 0 fixed = 2 total (was 1) |
| +1 | mvnsite | 27 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 750 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | the patch passed |
| +1 | findbugs | 56 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 82 | hadoop-azure in the patch passed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 3096 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-971/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/971 |
| JIRA Issue | HADOOP-16376 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 741bffa7ea39 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / e70aeb4 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-971/2/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-971/2/testReport/ |
| Max. process+thread count | 445 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-971/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
LGTM, +1

**Comment by DadanielZ** [comment]
Thanks for the review, closing this pr as it has been committed into trunk.

**Pull request closed without merging**